test/fluttertagger_test.dart

```diff
@@ -0,0 +1,272 @@
+              ),
+              controller: controller,
+              onSearch: onSearch,
+              builder: (context, key) {
+                return TextField(
+                  key: key,
+                  controller: controller,
+                  onChanged: (value) {},
+                );
+              },
+              triggerCharacterAndStyles: const {
+                '@': TextStyle(color: Colors.blue),
+                '#': TextStyle(color: Colors.green),
+              },
+            ),
+          ),
+        ),
+      );
+
+      await tester.pumpWidget(testWidget);
+
+      final textField = find.byType(TextField);
+      await tester.tap(textField);
+      await tester.pump();
+
+      // Incrementally enter each character to simulate typing
+      await tester.enterText(textField, '@');
+      await tester.pump();
+      await tester.enterText(textField, '@t');
+      await tester.pump();
+      await tester.enterText(textField, '@te');
+      await tester.pump();
+      await tester.enterText(textField, '@tes');
+      await tester.pump();
+      await tester.enterText(textField, '@test');
+      await tester.pumpAndSettle();
+
+      expect(searchQuery, 'test');
+      expect(triggerChar, '@');
+    });
+
+    testWidgets('displays overlay when typing a trigger character',
+        (tester) async {
+      await tester.pumpWidget(testWidget);
+
+      final textField = find.byType(TextField);
+      await tester.tap(textField);
+      await tester.pump();
+
+      // Incrementally enter each character to simulate typing
+      await tester.enterText(textField, '@');
+      await tester.pump();
+      await tester.enterText(textField, '@t');
+      await tester.pump();
+      await tester.enterText(textField, '@te');
+      await tester.pump();
+      await tester.enterText(textField, '@tes');
+      await tester.pump();
+      await tester.enterText(textField, '@test');
+      await tester.pumpAndSettle();
+
+      expect(find.text('Overlay'), findsOneWidget);
+    });
+
+    testWidgets('formats and displays tagged text correctly', (tester) async {
+      await tester.pumpWidget(testWidget);
+      await tester.enterText(find.byType(TextField), '@testUser');
+      await tester.pump();
+
+      expect(controller.text, '@testUser');
+    });
+
+    testWidgets('hides overlay when exiting search context', (tester) async {
+      await tester.pumpWidget(testWidget);
+      await tester.enterText(find.byType(TextField), '@test ');
+      await tester.pump();
+
+      expect(find.text('Overlay'), findsNothing);
+    });
+
+    testWidgets('handles nested tags correctly', (tester) async {
+      await tester.pumpWidget(testWidget);
+      await tester.enterText(find.byType(TextField), '@test @nestedTag');
+      await tester.pump();
+
+      expect(controller.text, '@test @nestedTag');
+    });
+
+    testWidgets('handles overlay positioning correctly', (tester) async {
```

Comment:
It's not clear what this test does differently than the `displays overlay when typing a trigger character` test.

Reply:
Updated the test

Reply:
@Mastersam07  can you add a test for OverlayPosition.top?

Reply:
Treated.